tests/assembly/asm/xtensa-types.rs

```diff
@@ -111,25 +110,25 @@ macro_rules! check_explicit_reg {
 
 // CHECK-LABEL: a5_i8:
 // CHECK: #APP
-// CHECK: mov a5, a5
+// CHECK: or a5, a5, a5
 // CHECK: #NO_APP
 check_explicit_reg!(a5_i8 i8 "a5" "mov");
```

Comment:
Btw, this test actually appears to be doing what the ISA says it should not do.
[Xtensa Instruction Set Architecture (ISA) Reference Manual](https://web.archive.org/web/20241005102231/https://0x04.net/~mwk/doc/xtensa.pdf) (436 page) says:

> `ar` and `as` should not specify the same register due to the `MOV.N` restriction.

Reply:
Interestingly the official ISA from cadence omits this line: https://www.cadence.com/content/dam/cadence-www/global/en_US/documents/tools/silicon-solutions/compute-ip/isa-summary.pdf (page 495). I'll ask the LLVM team what their reference is.

Reply:
Ah, it seems that what I was looking at was an old manual.

https://web.archive.org/web/20241005102231/https://0x04.net/~mwk/doc/xtensa.pdf (page 2)

> Issue Date: 4/2010
> RC-2010.1 Release
> PD-09-0801-10-01

https://www.cadence.com/content/dam/cadence-www/global/en_US/documents/tools/silicon-solutions/compute-ip/isa-summary.pdf (page 2)

> Product Release:RI-2021.8
> Last Updated:04/2022
> Modification: 737871